core/src/por/reporter.ts

```diff
@@ -27,10 +28,10 @@ async function shouldReport({
 
   // Check Submission Hearbeat
   const updatedAt = Number(latestRoundData.updatedAt) * 1000 // convert to milliseconds
-  const timestamp = Date.now()
+  const now = Date.now()
   const heartbeat = aggregator.heartbeat
 
-  if (updatedAt + heartbeat < timestamp) {
+  if (heartbeat >= POR_LATENCY_BUFFER && updatedAt + heartbeat - POR_LATENCY_BUFFER < now) {
```

Comment:
@bayram98 I think this does not help us to catch the error at the beginning. It will actually hide the issue until the time when we expect the heartbeat to occur.

`heartbeat >= POR_LATENCY_BUFFER` might evaluate to `false` and if there is no deviation change, then the function `shouldReport` returns false and behaves as if nothing wrong happened.

If `heartbeat < POR_LATENCY_BUFFER` is true, then the settings is wrongly defined, and we should throw an error.

```typescript
if (heartbeat < POR_LATENCY_BUFFER) {
    throw Error('Heartbeat cannot be smaller then latency buffer.')
}
```

Reply:
@martinkersner I added your suggestion to check if heartbeat and POR_LATENCY_BUFFER are set correctly or not. Please check it out.

Reply:
Beautiful! Let's merge!